__tests__/components/CommentsList.test.tsx

```diff
@@ -23,57 +23,89 @@ import { CommentCardProps } from "@/components/CommentCard";
 import {
   generateComment,
   generateNResults,
-  generatePagination,
 } from "@mocks/dprApplicationFactory";
 
 jest.mock("@/components/CommentCard", () => ({
   CommentCard: ({ comment, commentNumber }: CommentCardProps) => (
     <div data-testid="comment-card">
       <p>comment number: {commentNumber}</p>
-      <p>comment sentiment: {comment.sentiment}</p>
-      <p>comment: {comment.comment}</p>
+      <p>comment sentiment: {comment?.sentiment}</p>
+      <p>comment: {comment?.comment}</p>
     </div>
   ),
 }));
 
 describe("CommentsList", () => {
   it("shows correct public comments results", () => {
+    const comments = generateNResults<DprComment>(10, generateComment);
```

Comment:
we add a props to generateComments in the commentsCard, now it's a function: generateNResults<DprComment>(10, () => generateComment()). The default value is true